kong/plugins/correlation-id/handler.lua

```diff
@@ -76,8 +75,11 @@ function CorrelationIdHandler:header_filter(conf)
   end
 
   local correlation_id = kong.ctx.plugin.correlation_id or
-                         kong.request.get_header(conf.header_name) or
```

Comment:
Maybe we still need to keep this line, because if `kong.request.get_header(conf.header_name)` returned something in `access` phase, then the https://github.com/Kong/kong/pull/6358/files#diff-8e9c59879919fc95ad333be729b1ba4fR57 condition is not met and `kong.ctx.plugin.correlation_id` will therefore not be set.